### PR TITLE
Make attach_function signature more readable

### PR DESCRIPTION
**Pull request by pitr-ch**

### CODE REVIEW

**Comment by eregon** [comment]
This means we cannot just use this `attach_function` for rubysl-socket, without modifications to that code or having another `attach_function` doing the translation of arguments, and it's incompatible with http://www.rubydoc.info/github/ffi/ffi/FFI%2FLibrary%3Aattach_function

File: src/main/ruby/core/posix.rb [hunk 1]
```diff
@@ -161,29 +158,29 @@ def self.attach_function(method_name, native_name, argument_types, return_type =
...
+
   # Other routines
   attach_function :crypt, [:string, :string], :string
-  
+
   # Errno-related
 
   if Rubinius.linux?
-    attach_function :errno_address, :__errno_location, [], :pointer
+    attach_function :__errno_location, [], :pointer, as: :errno_address
```
Comment (eregon) [inline src/main/ruby/core/posix.rb]: One thing I don't like with this syntax is that it's unclear which method is getting defined, I need to read until the end to know it.

Comment (chrisseaton) [reply src/main/ruby/core/posix.rb]: I'm not sure that criticism makes sense. The previous approach had two names listed - it was completely unclear which method is getting defined. This approach which has an `as:` label that really does make it obvious.

**Comment by chrisseaton** [comment]
Could we support both approaches simultaneously? Or we can modify RubySL if we want?

File: src/main/ruby/core/posix.rb (hunk 1 above)
Comment (eregon) [reply src/main/ruby/core/posix.rb]: The first argument is always the name of the method getting defined, the rest is meta-information for binding the native function.

**Comment by eregon** [comment]
Let's merge this, since you both agree it makes the code clearer, and it simplifies the arguments handling.
I'll need to define some `attach_function` in rubysl-socket anyway, delegating to this one, and therefore that one will do the arguments conversion.

**Pull request merged**